### DISCUSSION TITLE
Crud default search

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#1102] [Q&A (Help)]
Hi all, i've a crud for this model:

```
class EmailTemplate extends Model
{
    use CrudTrait;
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'company_id',
        'template_id',
        'reason_id',
        'language',
        'subject',
        'content',
        'template_key',
        'importable'
    ];

    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'id' => 'integer',
    ];

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }

    public function templateType(): BelongsTo
    {
        return $this->belongsTo(TemplateType::class, 'template_id');
    }

    public function reason(): BelongsTo
    {
        return $this->belongsTo(Reason::class, 'reason_id');
    }

}
```
the result is this:
<img width="1273" alt="Screenshot 2024-08-01 alle 18 08 41" src="https://github.com/user-attachments/assets/a497275d-9297-4fab-8313-4ad5d85c7b2a">

Talking about this column:
```
        CRUD::addColumn([
            'name' => 'templateType.name',
            'label' => trans('backpack::EmailTemplate.template'),
            'type' => 'text',
            'limit' => 5000,
        ]);
```

I've seen that, by default the BKP Crud Search function does'nt search in this field..
So i've some questions:
1. By default, which fields does "BKP Search" search on?
2. What is the best method to enable the search also on fields of linked tables?
3. How do I put the classic "x" that deletes the search key?

Thanks for help, if there is still anyone not on holiday.. :)

**Comment by jcastroa87** [comment]
Hello @blondie63 

1. In the local field, in this case something like "template_id".

2. Did you try with "searchLogic"? Let me know if you have any problem.

3. You need to customize in "list.blade.php" the input on like a bootstrap input and a clear button, example:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Input Clear Button</title>
    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/5.3.0/css/bootstrap.min.css" rel="stylesheet">
    <style>
        .input-group-text {
            cursor: pointer;
        }
    </style>
</head>
<body>
    <div class="container mt-5">
        <div class="input-group">
            <input type="text" class="form-control" id="searchInput" placeholder="Type something...">
            <span class="input-group-text" id="clearInput">&times;</span>
        </div>
    </div>

    <script>
        document.getElementById('clearInput').addEventListener('click', function() {
            document.getElementById('searchInput').value = '';
        });
    </script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/5.3.0/js/bootstrap.bundle.min.js"></script>
</body>
</html>
```

Cheers.

**Comment by blondie63** [comment]
3. ok thanks !
2. I've just used it but i've some issues, maybe I'm naive but I thought that the search would work on all visible columns since the values ​​are already extracted and it wasn't necessary to do a specific SearchLogic!
1. I have done various tests and I have seen that the search of a CRUD without any custom SearchLogic searches on all visible columns as long as they come from the main table!
It does not search on the columns of relation fields
Regards

Mauro

**Comment by jcastroa87** [comment]
Hello @blondie63 do you have any error detail in which you can help or we can close this issue?

Cheers.

**Reply by blondie63** [reply]
do you mean about point N. 4 ?
